src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -185,6 +185,8 @@ internal bool ForceAffinityOutOfProc
         /// </summary>
         private string _noLoggingCompletedSubmissionDetails;
 
+        private static bool _isFirstWrite = true;
```

Comment:
We need to clarify that it is a first write of the debug file. Maybe `_debugDumpIsFirstWrite`? A comment above would be also nice to have.